viewpoints.php

```diff
@@ -0,0 +1,87 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * This file contains functions to edit observation points for an observation.
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require_once($dir . '../../config.php');
+
+$id = required_param('id', PARAM_INT); // Observation instance ID.
+$pointid = optional_param('pointid', null, PARAM_INT); // Observation point ID.
+$action = optional_param('action', null, PARAM_TEXT); // Action.
+
+list($observation, $course, $cm) = \mod_observation\manager::get_observation_course_cm_from_obid($id);
+
+// Check permissions.
+require_login($course, true, $cm);
+require_capability('mod/observation:editobservationpoints', $PAGE->context);
+
+$pageurl = new moodle_url('/mod/observation/viewpoints.php', array('id' => $id));
+
+// Check if action and pointid are present.
+if ($action !== null && $pointid !== null) {
+    if ($action == 'edit') {
```

Comment:
This could likely be a bit cleaner as a switch. Also a bit more performant (negligible)